app/src/main/java/org/jellyfin/androidtv/ui/search/SearchRepository.kt

```diff
@@ -0,0 +1,43 @@
+package org.jellyfin.androidtv.ui.search
+
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import org.jellyfin.androidtv.auth.repository.UserRepository
+import org.jellyfin.androidtv.util.apiclient.AsyncState
+import org.jellyfin.sdk.api.client.ApiClient
+import org.jellyfin.sdk.api.client.extensions.itemsApi
+import org.jellyfin.sdk.model.api.BaseItemKind
+import org.jellyfin.sdk.model.api.ItemFields
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+object SearchRepository: KoinComponent {
```

Comment:
This is not how our repositories work. We generally have an interface+implementation and do dependency injection via constructor.

Reply:
Updated, please ensure I did this correctly